lib/utilities/collectResponseHeaders.js

```diff
@@ -69,6 +69,14 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
         responseMetaHeaders['x-amz-replication-status'] =
             objectMD.replicationInfo.status;
     }
+    if (objectMD.replicationInfo &&
+        objectMD.replicationInfo.storageType &&
+        objectMD.replicationInfo.status === 'COMPLETED') {
+        const { storageType, dataStoreVersionId } = objectMD.replicationInfo;
+        const headerStorageType = storageType.replace('_', '-');
+        responseMetaHeaders[`x-amz-meta-${headerStorageType}-version-id`] =
```

Comment:
To confirm: this is the format we want for user metadata properties (where an object replicated to an external location with the type aws_s3 would have a Metadata field, for example: